expression/constant.go

```diff
@@ -450,3 +451,62 @@ func (c *Constant) Coercibility() Coercibility {
 	}
 	return c.collationInfo.Coercibility()
 }
+
+// TryUpdateCollate Update collate based on priority #35149
+func TryUpdateCollate(eq *ScalarFunction) {
+
+	if eq.FuncName.L != ast.EQ {
+		return
+	}
+	ok := expressionUpdateCollate(eq, true)
+	if !ok {
+		expressionUpdateCollate(eq, false)
+	}
+
+}
+
+// ExpressionUpdateCollate checks whether the expression can be updated to the new collation.
+func expressionUpdateCollate(eq *ScalarFunction, colIsLeft bool) bool {
+	var col *Column
+	var con *Constant
+	colOk := false
+	conOk := false
+	if colIsLeft {
```

Comment:
I think you can judge it by you self but not the input paramter "colIsLeft "

Reply:
I implemented this function imitating validEqualCond Helper